### ISSUE TITLE
IETF RFCs

### ISSUE THREAD

**Issue by jbenet** [ipfs/archives#18]
All RFCs! \o/

**Comment by eminence** [comment]
As a start, I've copied the .txt versions of all RFCs here:  

[QmNdozNckkKUAC2YfTLDMyASRsF8bKpyYgZrBR1Syiv2XK](http://ipfs.io/ipfs/QmNdozNckkKUAC2YfTLDMyASRsF8bKpyYgZrBR1Syiv2XK)

Using this hacky python script: [QmWg26A8LRR7tvHkKXRM7r9ZVPgAACudd5qs2LobjbtWaH](http://ipfs.io/ipfs/QmWg26A8LRR7tvHkKXRM7r9ZVPgAACudd5qs2LobjbtWaH)

**Comment by davidar** [comment]
@eminence Awesome, thankyou. Pinning now :)

**Comment by rht** [comment]
There is ftp link updated weekly here https://www.rfc-editor.org/download.html, which can be updated with rsync.

**Comment by davidar** [comment]
@rht yeah, we need an ipfs-rsync tool that can automate the update process, i.e.

```sh
$ ipfs-rsync ftp.rfc-editor.org::rfcs-text-only <old-hash>
Updated X files using Y MB.
Added to <new-hash>
```

**Comment by rht** [comment]
(That intersects with gx, dag versioning(, dat?))

**Comment by eminence** [comment]
Thanks for the pointer about the download bundle.  I was foolish and didn't even bother to look for it last night.  

The download bundle includes .pdf and .ps versions, which I've now included.  The new root is [QmeZnaoktRvs3Ek8pQNyj8BbrkKuJ5Ez6JHi32dazL7cEJ](http://ipfs.io/ipfs/QmeZnaoktRvs3Ek8pQNyj8BbrkKuJ5Ez6JHi32dazL7cEJ)

At this point, I'm not sure what the best way to mark the previous hash as obsolete (other than editing my github comment, which I've done)

**Comment by jbenet** [comment]
@eminence this is great!

**Comment by ion1** [comment]
Here’s a command that replicates the same directory structure:

```
% rsync -Pav --delete -f '+ /rfc-index.txt' -f '+ /rfc[0-9]*' -f '- *' ftp.rfc-editor.org::rfcs/ rfcs/ && ipfs add -r rfcs 
…
added Qmew9eTZSvPPJVZcv9WLdH7VJ9nKe4QVxxsi4iSLXNhkrq rfcs
```

(They have added three new RFCs since the previous pointer.)

**Comment by eminence** [comment]
Thanks @ion for that useful rsync command.  I've updated the RFC directory, and have published it under my peer name.  I will endeavor to keep it up-to-date.  

The new location for these is now the follow IPNS URL (which should remain unchanged as this folder is updated)

http://ipfs.io/ipns/QmbuG3dYjX5KjfAMaFQEPrRmTRkJupNUGRn1DXCgKK5ogD/archives/RFCs

**Comment by jbenet** [comment]
nice! we're in the middle of fixing ipns so you can have more keys, too

related to #5

**Comment by eminence** [comment]
Here's a simple front-end to the RFC archive

http://ipfs.io/ipfs/Qmee6stB1LMDBnguijenmhk9eMeqi1er8L5CjFiuut1LJL/frontend.html

**Comment by lgierth** [comment]
This is pretty cool! I thought there'd be no RFCs though, until I used the search :)

**Comment by rht** [comment]
It appears that the copyright license for the RFCs since nov 10 2008 is spec-ed in https://tools.ietf.org/html/rfc5378.

**Comment by leerspace** [comment]
The hash on archives.ipfs.io seems to have become unavailable (QmTZeE4yTiDd76YxbWvenneqSf5VhooNnA46WWWHtHcu5n). I'm not sure how it was created, so I re-ran the creation script that ion1 posted earlier in this thread and pinned it to my node (QmZHbZahTjnPTNXJaNnaVw5KusGuCL1EbbGp7qM4AxP4hG).

It doesn't fix the hash on the archives site, but now almost all of the files in QmeZnaoktRvs3Ek8pQNyj8BbrkKuJ5Ez6JHi32dazL7cEJ (posted by eminence) are available again; all except for the rfc-index.txt file.

**Comment by eminence** [comment]
Sorry about that.  my ipfs node had been offline for a few months due to hardware issues (which have now been fixed), but tonight i ran into troubles getting my ipfs daemon running (repo migration errors).  i'll continue to try to get it running again

**Comment by whyrusleeping** [comment]
@eminence We talked on IRC, but did you file an issue somewhere about that? I very much want to figure out the problem you were having

**Comment by eminence** [comment]
Sorry for the delay.   All issues are now resolved, and this archive is back online.
